run.py

```diff
@@ -12,7 +12,7 @@ def main(config: DictConfig):
     from src.utils import template_utils
 
     # load environment variables from `.env` file if it exists
-    dotenv.load_dotenv(dotenv_path=".env", override=True)
+    dotenv.load_dotenv(dotenv_path=hydra.utils.to_absolute_path(".env"), override=True)
```

Comment:
I now realised it's better to simply move this line outside of `main()`.

If we load env variables inside `main()` then we can't use environment variables inside our hydra specific config, e.g. if we would want to do something like this in `config.yaml`: 
```yaml
hydra:
    run:
        dir: ${env:MY_VAR}
```
it won't work because currently environment variables are loaded after hydra is loaded. Loading variables before `@hydra.main()` decorator will fix that. With that there's also no need to use `hydra.utils.to_absolute_path`

Reply:
You are right, put this load env procedure before `@hydra.main()` is right, But maybe the correct solution is `os.path.join(os.path.dirname(os.path.abspath(__file__)), ".env")`, since simply use ".env" have problem when run `python xxx/xxx/run.py`.

Reply:
True but it's generally a bad idea to run script outside of main folder - in such case the path to `data/` and `logs/` get place somewhere else. I tried to achieve placement of those folder to be always relative to run.py no matter from where we execute scipt, but it seems to not be possible with Hydra currently. 

Also I'd like to avoid importing any more dependencies outside of `main()` because it slows down tab completion significantly.

There is of course a use case when user specifies all the paths to logs, data and others in environment variables but I'm not sure it's worth covering for simplicity.

Reply:
Ok, I understanded what you say. In my case I need specify other log dir and run script outsize main folder, maybe we can add some explain in README to prevent careless use.

Reply:
I just looked at dotenv package and it turns out in your use case you can simply do:
```python
dotenv.load_dotenv(override=True)
```
If we don't specify path then dotenv recursively searches for file named `.env` in all higher folders. Tested it and it works.

Reply:
So I guess I will change the line to `dotenv.load_dotenv(override=True)` since it covers all use cases

Reply:
“recursively searches for file named .env in all higher folders”, does it works exactly this? I think search in higher folders  have not cover my case, since I run script in a higher folder, and .env file in a lower folder.

Reply:
I tested with this structure:
```python
├── template
│       ├── run.py
│
├──  .env
```
`python template/run.py`

Seems to work correctly. Does this cover your use case?

Reply:
Oh, I thought into wrong way. It's right I can always use it safety.